test/s3select_test.cpp

```diff
@@ -238,3 +238,75 @@ TEST(TestS3SElect, utcnow_function)
     ASSERT_EQ(s3select_res, boost::posix_time::to_simple_string(now));
 }
 
+void generate_csv(std::string& out, size_t size) {
+  // schema is: int, float, string, string
+  std::stringstream ss;
+  for (auto i = 0U; i < size; ++i) {
+    ss << i << "," << i/10.0 << "," << "foo"+std::to_string(i) << "," << std::to_string(i)+"bar" << std::endl;
+  }
+  out = ss.str();
+}
+
+TEST(TestS3selectFunctions, mod)
+{
+    s3select s3select_syntax;
+    const std::string input_query = "select 5%2 from stdin;";
+    auto status = s3select_syntax.parse_query(input_query.c_str());
+    ASSERT_EQ(status, 0);
+    s3selectEngine::csv_object s3_csv_object(&s3select_syntax);
+    std::string s3select_result;
+    std::string input;
+    size_t size = 1;
+    generate_csv(input, size);
+    status = s3_csv_object.run_s3select_on_object(s3select_result, input.c_str(), input.size(), 
+        false, // dont skip first line 
+        false, // dont skip last line
+        true   // aggregate call
+        ); 
+    ASSERT_EQ(status, 0);
+    ASSERT_EQ(s3select_result, std::string("1,\n"));
+}
+
+TEST(TestS3selectFunctions, modzero)
+{
+    s3select s3select_syntax;
+    const std::string input_query = "select 0%2 from stdin;";
+    auto status = s3select_syntax.parse_query(input_query.c_str());
+    ASSERT_EQ(status, 0);
+    s3selectEngine::csv_object s3_csv_object(&s3select_syntax);
+    std::string s3select_result;
+    std::string input;
+    size_t size = 1;
+    generate_csv(input, size);
+    status = s3_csv_object.run_s3select_on_object(s3select_result, input.c_str(), input.size(), 
+        false, // dont skip first line 
+        false, // dont skip last line
+        true   // aggregate call
+        ); 
+    ASSERT_EQ(status, 0);
+    ASSERT_EQ(s3select_result, std::string("0,\n"));
+}
+
+TEST(TestS3selectFunctions, modnegative)
+{
+    s3select s3select_syntax;
+    const std::string input_query = "select -13%5 from stdin;";
+    auto status = s3select_syntax.parse_query(input_query.c_str());
+    ASSERT_EQ(status, 0);
+    s3selectEngine::csv_object s3_csv_object(&s3select_syntax);
+    std::string s3select_result;
+    std::string input;
+    size_t size = 1;
+    generate_csv(input, size);
+    status = s3_csv_object.run_s3select_on_object(s3select_result, input.c_str(), input.size(), 
+        false, // dont skip first line 
+        false, // dont skip last line
+        true   // aggregate call
+        ); 
+    ASSERT_EQ(status, 0);
+    ASSERT_EQ(s3select_result, std::string("2,\n"));
```

Comment:
the result should be negative,no?

Reply:
No, It's positive in this case(-13 % 5) It's negative on (13 % -5)